pkg/server/server_test.go

```diff
@@ -49,23 +50,42 @@ func TestStringEncode(t *testing.T) {
 	}
 }
 
-func TestMachineConfigToIgnition(t *testing.T) {
+func TestMachineConfigToRawIgnition(t *testing.T) {
 	mcPath := filepath.Join(testDir, "machine-configs", testConfig+".yaml")
 	mcData, err := ioutil.ReadFile(mcPath)
 	assert.Nil(t, err)
 	mc := new(mcfgv1.MachineConfig)
 	err = yaml.Unmarshal([]byte(mcData), mc)
 	assert.Nil(t, err)
-	assert.Equal(t, 1, len(mc.Spec.Config.Storage.Files))
-	assert.Equal(t, mc.Spec.Config.Storage.Files[0].Path, "/etc/coreos/update.conf")
+	mcIgnCfg, _, err := ign.Parse(mc.Spec.Config.Raw)
+	if err != nil {
+		t.Errorf("decoding Ignition Config failed: %s", err)
+	}
```

Comment:
(These `if err != nil` checks can be `{assert,require}.NoError(t, err)` . OTOH, existing style in parts of this file…)

Reply:
yeh, I was trying to adhere to existing style :)